File: app/src/main/java/com/liushu/crazyandroid/ui/main/activity/SplashActivity.java
package com.liushu.crazyandroid.ui.main.activity;

import android.content.Intent;
import android.view.View;
import android.widget.Button;

import com.jaydenxiao.common.base.BaseActivity;
import com.liushu.crazyandroid.R;
import com.liushu.crazyandroid.adapter.TextTagsAdapter;
import com.liushu.crazyandroid.adapter.VectorTagsAdapter;
import com.liushu.crazyandroid.adapter.ViewTagsAdapter;
import com.moxun.tagcloudlib.view.TagCloudView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.OnClick;

public class SplashActivity extends BaseActivity {

    @BindView(R.id.tag_cloud)
    TagCloudView mTagCloud;
    @BindView(R.id.tag_text)
    Button mTagText;
    @BindView(R.id.tag_view)
    Button mTagView;
    @BindView(R.id.tag_vector)
    Button mTagVector;

    private TextTagsAdapter textTagsAdapter;
    private ViewTagsAdapter viewTagsAdapter;
    private VectorTagsAdapter vectorTagsAdapter;

    private List<String> mStrings;

    @Override
    public int getLayoutId() {
        return R.layout.activity_splash;
    }

    @Override
    public void initView() {
        mStrings = new ArrayList<>();
        for (int i = 1; i < 17; i++) {
            mStrings.add("第" + i + "章");
        }
        textTagsAdapter = new TextTagsAdapter(mStrings);
        textTagsAdapter.setListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (v.getTag() instanceof Integer) {
                    int position = (int) v.getTag();
                    Intent intent = new Intent(mContext, MainActivity.class);
                    intent.putExtra("position", position);
                    startActivity(intent);
                    finish();
                }
            }
        });
        viewTagsAdapter = new ViewTagsAdapter();
        vectorTagsAdapter = new VectorTagsAdapter();

        mTagCloud.setAdapter(textTagsAdapter);
    }

    @OnClick({R.id.tag_text, R.id.tag_view, R.id.tag_vector})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.tag_text:
                mTagCloud.setAdapter(textTagsAdapter);
                break;
            case R.id.tag_view:
                mTagCloud.setAdapter(viewTagsAdapter);
                break;
            case R.id.tag_vector:
                mTagCloud.setAdapter(vectorTagsAdapter);
                break;
        }
    }
}

File: app/src/main/java/com/liushu/crazyandroid/ui/main/activity/SplashActivity.kt
package com.liushu.crazyandroid.ui.main.activity

import android.content.Intent
import android.view.View
import butterknife.OnClick
import com.jaydenxiao.common.base.BaseActivity
import com.liushu.crazyandroid.R
import com.liushu.crazyandroid.adapter.TextTagsAdapter
import com.liushu.crazyandroid.adapter.VectorTagsAdapter
import com.liushu.crazyandroid.adapter.ViewTagsAdapter
import kotlinx.android.synthetic.main.activity_splash.*
import java.util.*

class SplashActivity : BaseActivity() {

    private var textTagsAdapter: TextTagsAdapter? = null
    private var viewTagsAdapter: ViewTagsAdapter? = null
    private var vectorTagsAdapter: VectorTagsAdapter? = null

    private var mStrings: MutableList<String>? = null

    override val layoutId: Int
        get() = R.layout.activity_splash

    override fun initView() {
        mStrings = ArrayList()
        for (i in 1..16) {
            mStrings!!.add("第" + i + "章")
        }
        textTagsAdapter = TextTagsAdapter(mStrings)
        textTagsAdapter!!.setListener { v ->
            if (v.tag is Int) {
                val position = v.tag as Int
                val intent = Intent(mContext, MainActivity::class.java)
                intent.putExtra("position", position)
                startActivity(intent)
                finish()
            }
        }
        viewTagsAdapter = ViewTagsAdapter()
        vectorTagsAdapter = VectorTagsAdapter()

        tag_cloud!!.setAdapter(textTagsAdapter)
    }

    @OnClick(R.id.tag_text, R.id.tag_view, R.id.tag_vector)
    fun onViewClicked(view: View) {
        when (view.id) {
            R.id.tag_text -> tag_cloud!!.setAdapter(textTagsAdapter)
            R.id.tag_view -> tag_cloud!!.setAdapter(viewTagsAdapter)
            R.id.tag_vector -> tag_cloud!!.setAdapter(vectorTagsAdapter)
        }
    }
}
